src/adapter/src/catalog.rs

```diff
@@ -6190,33 +5979,22 @@ pub enum SerializedCatalogItem {
 
 /// Serialized (stored alongside the replica) logging configuration of
 /// a replica. Serialized variant of `ReplicaLogging`.
-///
-/// A `None` value for `sources` or `views` indicates that we did not yet create them in the
-/// catalog. This is only used for initializing the default replica of the default cluster.
-///
-/// To indicate the absence of sources/views, use `Some(Vec::new())`.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
 pub struct SerializedReplicaLogging {
     log_logging: bool,
     interval: Option<Duration>,
-    sources: Option<Vec<(LogVariant, GlobalId)>>,
-    views: Option<Vec<(LogView, GlobalId)>>,
```

Comment:
This is a change to the serialized catalog state, but according to my testing the deserialization ignores unknown fields, which makes a migration unnecessary. We could still write one to proactively clean up the serialized state. What do the surfaces folk think?

Reply:
I think as long as we don't add ones with the same names, we should be fine? Just guessing. But you are right, better safe than sorry. I'll add a migration.

Reply:
@jkosh44 Do you know how to write such a migration? I tried:
```diff
diff --git a/src/adapter/src/catalog/storage.rs b/src/adapter/src/catalog/storage.rs
index edf681b33d..2beaa21275 100644
--- a/src/adapter/src/catalog/storage.rs
+++ b/src/adapter/src/catalog/storage.rs
@@ -385,6 +385,16 @@ async fn migrate(
             )?;
             Ok(())
         },
+        // The fields relating to persisted introspection sources and views
+        // where removed from the replica config.
+        //
+        // Introduced in v0.48.0
+        //
+        // TODO(jkosh44) Can be cleared (patched to be empty) v0.50.0
+        |txn: &mut Transaction<'_>, _now, _bootstrap_args| {
+            txn.cluster_replicas.update(|_k, v| Some(v.clone()))?;
+            Ok(())
+        }
         // Add new migrations above.
         //
         // Migrations should be preceded with a comment of the following form:
```

But that doesn't work, the JSON stored in stash remains unchanged. AFAICT this is because `update` inserts a -1 update for the old `v` and a +1 update for the new `v`. It then serializes both `v`s to JSON and adds them to a stash batch. So in the end that batch has updates inserting and retracting the same value, while the actual old JSON value is untouched.

It seems like we need a way to migrate the serialized stash contents, but I don't think that exists.

Reply:
I naively thought that would have worked. Did you try deleting and then re-inserting and see if it's any better? Otherwise, let's just skip it for now.

Reply:
I tried that, it has the same effect. The underlying issue is that you can't tell stash to "delete the value at key `k`" but only "delete this (key, JSON value) pair". And since the migration works on deserialized values, it doesn't know how the old JSON value looked like.